src/IR/IR.hs

```diff
@@ -33,9 +42,23 @@ data Program t = Program
   { programEntry :: VarId
   , programDefs  :: [(VarId, Expr t)]
   , typeDefs     :: [(TConId, TypeDef t)]
+  , classDefs    :: [ClassDef t]
```

Comment:
let's use a more uniform naming convention. How about `programDefs`, `programClasses`, `programInsts`, and `programTypes`?